fix(overflow-menu): increase width of the spacer to match button width

kevinsperrine opened the pull request:
Closes #3013 #1885 

**Summary**

- The small spacer tab in the overflow menu wasn't quite long enough to fill the space leaving a small gap. This fixes that.

**Change List (commits, features, bugs, etc)**

- increase width of spacer tab to match button width

**Acceptance Test (how to verify the PR)**

- Go to [StatefulTable with row action](https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-statefultable--stateful-example-with-expansion-max-pages-and-column-resize) or [OverflowMenu stories](https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-overflowmenu--basic)
- open the overflow menu
- the spacer tab should match the width of the button

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Check other stories using overflow menu (in RTL) to confirm the fixes don't break anything
- https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--table-example-with-create-save-views
- https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--basic-table-with-full-row-edit-example
- https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-overflowmenu--basic

<!-- For reviewers: do not remove -->

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

bjornalm reviewed (approved):
Nice, work. This is not from your code changes, but it is related so perhaps we can fix it with this PR as well (or a new one of you want):

![image](https://user-images.githubusercontent.com/5167091/140524001-cb619524-5ab5-4234-b7f6-4d9c05704715.png)

kevinsperrine commented:
> Nice, work. This is not from your code changes, but it is related so perhaps we can fix it with this PR as well (or a new one of you want):
> 
> ![image](https://user-images.githubusercontent.com/5167091/140524001-cb619524-5ab5-4234-b7f6-4d9c05704715.png)

I actually think this is a carbon bug in this version, but I added a fix to my auto-positioning hook to account for it, and replaced the overflow menu in TableActionRow with the auto-positioning version (as requested in #1885), so I've re-requested a review for those changes.

bjornalm commented:
Code looks good but the  build failed so I the regression test links point to the previous version

bjornalm commented on packages/react/src/hooks/usePopoverPositioning.jsx:
```diff
@@ -116,6 +118,12 @@ export const usePopoverPositioning = ({
+      let left = 0;
+
+      if (langDir === 'rtl' && isOverflowMenu) {
+        left = -15;
```
Put this value 15 in a well named const that explains the origin?

kevinsperrine replied:
It's not necessary. Carbon has fixed it and I need to take it out in the carbon upgrade PR...

kevinsperrine replied:
Actually, nvm. It's not. I'll add a constant.

bjornalm commented:
I approved this, but I also found another related issue that also exists in next. 
![image](https://user-images.githubusercontent.com/5167091/141077476-ed70a4c3-effa-4e31-b130-374e72593bc9.png)

kevinsperrine commented:
> I approved this, but I also found another related issue that also exists in next. ![image](https://user-images.githubusercontent.com/5167091/141077476-ed70a4c3-effa-4e31-b130-374e72593bc9.png)

Yeah, there's already an issue for this. https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2951
